async-nats/src/service/mod.rs

```diff
@@ -365,36 +350,78 @@ async fn verb_subscription(
     Ok(stream::select_all([verb_all, verb_id, verb_name]).fuse())
 }
 
-impl Stream for Service {
+pub struct Endpoint {
+    requests: Subscriber,
+    stats: Arc<Mutex<Stats>>,
+    client: Client,
+    endpoint: String,
+    shutdown: Pin<Box<dyn Future<Output = Result<(), tokio::sync::broadcast::error::RecvError>>>>,
```

Comment:
Why not take the broadcast receiver here?

```suggestion
    shutdown: Pin<Box<dyn Future<Output = Result<(), tokio::sync::broadcast::error::RecvError>>>>,
```